align/pnr/main.py

```diff
@@ -341,16 +345,22 @@ def generate_pnr(topology_dir, primitive_dir, pdk_dir, output_dir, subckt, *, pr
                            'nvariants': nvariants,
                            'effort': effort}
 
-        top_level, leaf_map, placement_verilog_alternatives, metrics = \
-            placer_driver(cap_map=cap_map, cap_lef_s=cap_lef_s,
-                          lambda_coeff=lambda_coeff, scale_factor=scale_factor,
-                          select_in_ILP=select_in_ILP, place_using_ILP=place_using_ILP, seed=seed,
-                          use_analytical_placer=use_analytical_placer, ilp_solver=ilp_solver, primitives=primitives,
-                          toplevel_args_d=toplevel_args_d, results_dir=None,
-                          placer_sa_iterations=placer_sa_iterations, placer_ilp_runtime=placer_ilp_runtime)
+        if placer == 'python':
```

Comment:
@stevenmburns , this is where the Python placer is called. I am dumping pva just for observability. Its structure is as follows (truncated for readability):
```python
{
  "leaves": [
    {
      "abstract_name": "NMOS_3T_85679509",
      "concrete_name": "NMOS_3T_85679509_X2_Y2",
      "constraints": [],
      "bbox": [],
      "terminals": []
    },
    {
      "abstract_name": "DCL_PMOS_2T_69165982",
      "concrete_name": "DCL_PMOS_2T_69165982_X2_Y2",
      "constraints": [],
      "bbox": [],
      "terminals": []
    }
  ],
  "modules": [
    {
      "parameters": [
        "VIN",
        "VOP"
      ],
      "constraints": [],
      "instances": [
        {
          "instance_name": "X_MP0",
          "fa_map": [],
          "abstract_template_name": "DCL_PMOS_2T_69165982",
          "transformation": {
            "oX": 15120,
            "oY": 37800,
            "sX": -1,
            "sY": -1
          },
          "concrete_template_name": "DCL_PMOS_2T_69165982_X2_Y2"
        },
        {
          "instance_name": "X_MN0",
          "fa_map": [],
          "abstract_template_name": "NMOS_3T_85679509",
          "transformation": {
            "oX": 15120,
            "oY": 0,
            "sX": -1,
            "sY": 1
          },
          "concrete_template_name": "NMOS_3T_85679509_X2_Y2"
        }
      ],
      "global_signals": [
        "VCCX",
        "VSSX"
      ],
      "bbox": [],
      "abstract_name": "CKT_COMMON_SOURCE",
      "concrete_name": "CKT_COMMON_SOURCE_0",
      "metrics": {
        "cost": 73270.0,
        "width": 15120,
        "height": 37800,
        "area": 571536000,
        "hpwl": 40700.0
      }
    }
  ]
}
```